app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,67 @@
+require 'date'
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(completion_date: :desc)
+  end
+
+  def show
+    id = params[:id].to_i
+    @task = Task.find_by(id: id)
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id].to_i)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.